compliance/cis_k8s/rules/cis_4_2_12/rule.rego

```diff
@@ -0,0 +1,48 @@
+package compliance.cis_k8s.rules.cis_4_2_12
+
+import data.compliance.cis_k8s
+import data.compliance.lib.common
+import data.compliance.lib.data_adapter
+
+default rule_evaluation = false
+
+process_args := cis_k8s.data_adapter.process_args
+
+# Verify that the RotateKubeletServerCertificate argument is set to true
+rule_evaluation {
+	common.contains_key_with_value(process_args, "--feature-gates", "RotateKubeletServerCertificate=true")
+}
+
+# In case both flags and configuration file are specified, the executable argument takes precedence.
+rule_evaluation {
+	not process_args["--feature-gates"]
+	data_adapter.process_config.config.featureGates.RotateKubeletServerCertificate
+}
+
+rule_evaluation {
+	not contains(process_args["--feature-gates"], "RotateKubeletServerCertificate")
+	data_adapter.process_config.config.featureGates.RotateKubeletServerCertificate
+}
+
+rule_evaluation {
+    common.contains_key_with_value(process_args,"--rotate-server-certificates", "true")
+}
+
+rule_evaluation {
+    data_adapter.process_config.config.serverTLSBootstrap
+}
+
+# Verify that the RotateKubeletServerCertificate argument is set to true
```

Comment:
nip: we usually put this only at the top of the rule, It doesn't hurt to have it here also though